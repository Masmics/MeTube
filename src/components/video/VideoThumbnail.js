import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function VideoThumbnail({ id, url }) {
  return (
    <Link to={`/${id}`}>
      <img src={url} />
    </Link>
  );
}

VideoThumbnail.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default VideoThumbnail;
