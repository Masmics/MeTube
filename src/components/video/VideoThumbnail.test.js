import React from 'react';
import { shallow } from 'enzyme';
import VideoThumbnail from './VideoThumbnail';

describe('video thumbnail component', () => {
  it('renders a thumbnail image', () => {
    const wrapper = shallow(<VideoThumbnail id="-UoeQOC-5iw" url={'https://img.youtube.com/vi/-UoeQOC-5iw/0.jpg'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
