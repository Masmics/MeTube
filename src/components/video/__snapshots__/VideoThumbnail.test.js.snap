// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video thumbnail component renders a thumbnail image 1`] = `
<Link
  to="/-UoeQOC-5iw"
>
  <img
    src="https://img.youtube.com/vi/-UoeQOC-5iw/0.jpg"
  />
</Link>
`;
